src/librustc/ty/layout.rs

```diff
@@ -151,9 +189,9 @@ pub enum Endian {
 }
 
 /// Size of a type in bytes.
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub struct Size {
-    raw: u64
+    pub bytes: u64
```

Comment:
I actually initially wanted to use the highest bit to indicate that the rest of the size is the base size of an unsized type, instead of an exact size, but I ended up dealing with unsized types differently.
I could go back to private `raw` and add more checks for 2^61 overflow if you want to.